crates/oxc_transformer/src/regexp/mod.rs

```diff
@@ -0,0 +1,237 @@
+
+mod options;
+
+use std::borrow::Cow;
+
+pub use options::RegExpOptions;
+use oxc_ast::ast::*;
+use oxc_regular_expression::ast::{LookAroundAssertionKind, Pattern, Term};
+use oxc_semantic::ReferenceFlags;
+use oxc_traverse::Traverse;
+
+use crate::context::Ctx;
+
+pub struct RegExp<'a> {
+    ctx: Ctx<'a>,
+    options: RegExpOptions,
+}
+
+impl<'a> RegExp<'a> {
+    pub fn new(options: RegExpOptions, ctx: Ctx<'a>) -> Self {
+        Self { ctx, options }
+    }
+}
+
+impl<'a> Traverse<'a> for RegExp<'a> {
+    fn enter_expression(
+        &mut self,
+        expr: &mut Expression<'a>,
+        ctx: &mut oxc_traverse::TraverseCtx<'a>,
+    ) {
+        let Expression::RegExpLiteral(regexp) = expr else {
+            return;
+        };
+
+        let is_unsupported = self.has_unsupported_regular_expression_flags(regexp.regex.flags);
+
+        let mut pattern_source = Cow::Borrowed(match regexp.regex.pattern {
+            RegExpPattern::Raw(pattern)
+                if is_unsupported
+                    || self.has_unsupported_regex_syntax_raw(pattern, regexp.regex.flags) =>
+            {
+                pattern
+            }
+            RegExpPattern::Pattern(ref pattern)
+                if is_unsupported || self.has_unsupported_regular_expression_pattern(pattern) =>
+            {
+                regexp.regex.pattern.source_text(self.ctx.source_text)
+            }
+            _ => return,
+        });
+
+        if pattern_source.contains('\\') {
+            // Escape backslashes in the pattern source
+            pattern_source = Cow::Owned(pattern_source.replace('\\', "\\\\"));
+        }
+
+        let callee = {
+            let symbol_id = ctx.scopes().find_binding(ctx.current_scope_id(), "RegExp");
+            let ident = ctx.create_reference_id(
+                regexp.span,
+                "RegExp".into(),
+                symbol_id,
+                ReferenceFlags::read(),
+            );
+            ctx.ast.expression_from_identifier_reference(ident)
+        };
+
+        let mut arguments = ctx.ast.vec_with_capacity(2);
+        arguments.push(
+            ctx.ast.argument_expression(
+                ctx.ast.expression_string_literal(regexp.span, pattern_source),
+            ),
+        );
+
+        let flags = regexp.regex.flags.to_string();
+        let flags =
+            ctx.ast.argument_expression(ctx.ast.expression_string_literal(regexp.span, flags));
+        arguments.push(flags);
+
+        *expr = ctx.ast.expression_new(
+            regexp.span,
+            callee,
+            arguments,
+            None::<TSTypeParameterInstantiation>,
+        );
+    }
+}
+
+impl<'a> RegExp<'a> {
+    /// Check if the regular expression contains any unsupported flags.
+    fn has_unsupported_regular_expression_flags(&mut self, flags: RegExpFlags) -> bool {
+        flags.iter().any(|f| match f {
+            RegExpFlags::G | RegExpFlags::I | RegExpFlags::M => true,
```

Comment:
Shouldn't this be `false`? i.e. `g`, `i` and `m` flags are always supported?

Reply:
Yes, it is a mistake was made by me